torchgeo/trainers/segmentation.py

```diff
@@ -230,9 +324,14 @@ def training_step(
         batch_size = x.shape[0]
         y_hat = self(x)
         loss: Tensor = self.criterion(y_hat, y)
-        self.log('train_loss', loss, batch_size=batch_size)
+        self.log('train_loss', loss, batch_size=batch_size, on_epoch=True, on_step=True)
         self.train_metrics(y_hat, y)
-        self.log_dict(self.train_metrics, batch_size=batch_size)
+        self.log_dict(
+            {f'{k}': v for k, v in self.train_metrics.compute().items()},
```

Comment:
I'm a bit unsure about this workflow - presumably `compute` is called once at the `on_epoch` and the `reset` is called automatically after? Could be safer to use the `on_stage_epoch_end` unless anyone is very confident about this